### PR TITLE
DAOS-10175 test: add multi-threaded eq poller tests

### PR DESCRIPTION
**Pull request by mchaarawi**
- add unit test using only event API
- add similar test using DFS IO

Quick-Functional: true
Test-tag: test_daos_dfs_unit unittest

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/api/tests/eq_tests.c [hunk 1]
```diff
@@ -1043,6 +1043,145 @@ eq_test_8()
...
+			D_MUTEX_UNLOCK(&eqh_mutex);
+		}
+	}
+}
+
+static int
+eq_test_9()
+{
+	struct daos_event	*events[EQT_EV_COUNT] = { 0 };
+	struct daos_event	*eps[EQT_EV_COUNT] = { 0 };
```
Comment (liuxuezhao) [inline src/client/api/tests/eq_tests.c:1080]: EQT_EV_COUNT is defined as 1000, maybe not enough to test possible race condition. For larger number, may need to allocate memory.

**Review by liuxuezhao — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]
Pls address the merge conflict.

**Comment by mchaarawi** [comment]
> Pls address the merge conflict.

done

File: src/client/api/tests/eq_tests.c (hunk 1 above)
Comment (mchaarawi) [reply src/client/api/tests/eq_tests.c:1080]: yea sure, we can have a different PR for that. we've seen issues before on VM runs when the workload is high, so we can do a test to increase that there.

**Pull request merged**